lib/_http_client.js

```diff
@@ -30,11 +30,18 @@ function ClientRequest(options, cb) {
 
   var agent = options.agent;
   var defaultAgent = options._defaultAgent || Agent.globalAgent;
+
   if (agent === false) {
     agent = new defaultAgent.constructor();
-  } else if ((agent === null || agent === undefined) &&
-             typeof options.createConnection !== 'function') {
-    agent = defaultAgent;
+  } else if (agent == null) {
```

Comment:
@sam-github at your convenience could you check the comment. Wasn't too sure on the wording so it seems a little verbose to me given i was struggling on explaining the case. I'm open to rewrite.  Thanks again for your feedback.